src/ResourceManagement/CosmosDB/CosmosDBAccountImpl.cs

```diff
@@ -455,9 +461,17 @@ public IEnumerable<ISqlDatabase> ListSqlDatabases()
 
         public async Task<IEnumerable<ISqlDatabase>> ListSqlDatabasesAsync(CancellationToken cancellationToken = default(CancellationToken))
         {
-            var result = await this.Manager.Inner.SqlResources
-                .ListSqlDatabasesAsync(this.ResourceGroupName, this.Name);
-            return result.Select(inner => new SqlDatabaseImpl(inner));
+            return await this.sqlDatabases.ListAsync(cancellationToken);
+        }
+
+        public ISqlDatabase GetSqlDatabase(string databaseName)
+        {
+            return Extensions.Synchronize(() => this.GetSqlDatabaseAsync(databaseName));
+        }
+
+        public async Task<ISqlDatabase> GetSqlDatabaseAsync(string databaseName, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            return await this.sqlDatabases.GetImplAsync(databaseName, cancellationToken);
         }
```

Comment:
Should this Get (and List as well) go the `SqlDatabasesImpl sqlDatabases` now (since I think there is cached)?

Reply:
Actually the async function all use online request.

Reply:
Please double check the cache logic (it should be able to be invalidated to allow update, e.g. by a Refresh on child or on parent). If cache is good I think it is preferrable to use cache. @yaohaizh Let us know your opinion.

Reply:
Change to lazy load and refresh in cosmos db account.